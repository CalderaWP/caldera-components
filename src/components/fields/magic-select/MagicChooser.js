import React from 'react';
import {optionsShapeProp} from '../propTypes';
import PropTypes from 'prop-types'
/**
 * Creates a select field
 *
 * @param {Object} props
 * @returns {*}
 * @constructor
 */
export const MagicChooser = (props) => {
	/**
	 * Dispatches new value(s) to parent
	 *
	 * @param selection
	 * @returns {*}
	 */
	function changeHandler(selection) {
		return props.onMagicChange(selection);
	}

	return (
		<div
			role="group"
		>
			{props.options.map(option => {
				return <button
					key={option.value}
					onClick={changeHandler(option.value)}
				>
					{option.label}
				</button>
			})}
		</div>



	);
};

/**
 * Prop definition for select fields
 *
 * @type {{}}
 */
MagicChooser.propTypes = {
	onMagicChange: PropTypes.func.isRequired,
	options: optionsShapeProp
};

/**
 * Default props for select fields
 *
 * @type {{options: Array}}
 */
MagicChooser.defaultProps = {
	options: []
};