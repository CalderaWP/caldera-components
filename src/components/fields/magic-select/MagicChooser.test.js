import {MagicChooser} from './MagicChooser';
import renderer from 'react-test-renderer';
import React from 'react';
import {mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

describe('Select field component', () => {
	function genericChangeHandler() {

	}

	it('Creates with one option', () => {
		const component = renderer.create(
			<MagicChooser
				onMagicChange={genericChangeHandler}
				options={[
					{
						value: 'Tags',
						label: 'Tags'
					}
				]}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Sends the right update value when clicked', () => {
		let updateValue = null;
		const component = mount(
			<MagicChooser
				onMagicChange={(newValue) => {
					updateValue = newValue
				}}
				options={[
					{
						value: 'Tags',
						label: 'Tags'
					}
				]}
			/>
		);
		component.find('button').simulate('click');
		expect(updateValue).toBe('Tags')
	});


});