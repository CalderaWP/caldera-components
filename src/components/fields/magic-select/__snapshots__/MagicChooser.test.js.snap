// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select field component Creates with one option 1`] = `
<div
  role="group"
>
  <button
    onClick={undefined}
  >
    Tags
  </button>
</div>
`;
